hedera-node/hedera-evm-api/src/main/java/com/hedera/services/evm/contracts/execution/PricesAndFeesProvider.java

```diff
@@ -15,9 +15,22 @@
  */
 package com.hedera.services.evm.contracts.execution;
 
+import com.hederahashgraph.api.proto.java.ExchangeRate;
+import com.hederahashgraph.api.proto.java.FeeData;
 import com.hederahashgraph.api.proto.java.HederaFunctionality;
+import com.hederahashgraph.api.proto.java.Timestamp;
 import java.time.Instant;
 
 public interface PricesAndFeesProvider {
+    FeeData defaultPricesGiven(HederaFunctionality function, Timestamp at);
+
+    ExchangeRate rate(Timestamp at);
+
+    long estimatedGasPriceInTinybars(HederaFunctionality function, Timestamp at);
+
+    //    FeeObject estimatePayment(Query query, FeeData usagePrices, StateView view, Timestamp at,
```

Comment:
Since precompile related price logic would be added later, you can delete these methods from the interface and they could be added later.